src/mgr/DaemonServer.cc

```diff
@@ -1773,24 +1773,27 @@ void DaemonServer::got_mgr_map()
   Mutex::Locker l(lock);
   set<std::string> have;
   cluster_state.with_mgrmap([&](const MgrMap& mgrmap) {
+      auto md_update = [&] (DaemonKey key) {
+        std::ostringstream oss;
+        auto c = new MetadataUpdate(daemon_state, key);
+        oss << "{\"prefix\": \"mgr metadata\", \"who\": \""
+              << key.second << "\"}";
+        monc->start_mon_command({oss.str()}, {}, &c->outbl, &c->outs, c);
+      };
       if (mgrmap.active_name.size()) {
 	DaemonKey key("mgr", mgrmap.active_name);
 	have.insert(mgrmap.active_name);
 	if (!daemon_state.exists(key)) {
```

Comment:
I think on the osd/mds path we do some checking of whether the metadata is already being fetched in DaemonStateIndex.updating, should do similar here to avoid firing off multiple requests